pwndbg/dbg/__init__.py

```diff
@@ -12,6 +13,169 @@
 dbg: Debugger = None
 
 
+class TypeCode(Enum):
+    """
+    Broad categories of types.
+    """
+
+    POINTER = 1
+    ARRAY = 2
+    STRUCT = 3
+    TYPEDEF = 4
+    UNION = 5
+    INT = 6
+    ENUM = 7
+
+
+class TypeField:
```

Comment:
nit: I personally like `Field` as a name more than `TypeField`, but don't feel too strongly about it.

Reply:
LLDB has the concept of children values. I can't confirm it right now, but they seem to be [much like fields (see the registers example)](https://lldb.llvm.org/python_api/lldb.SBValue.html#lldb.SBValue.children), but for values, rather than types

I feel this can muddy the waters a little depending on how we want to treat them in the future, so I felt like specifying these fields specifically belong to a type would be clearer - in the sense that it communicates that they don't have an innate value, unlike ones from a value

I feel like a sensible choice if we end up exposing those would be to call them ValueFields, but, then again, we could just copy LLDB and call them "children"

That being said, I do recognize it's mostly a matter of taste, and I don't feel very strongly about the name we're going with here, either, so if you still feel it makes sense to change it to just Field, I'll do it

Reply:
Your reasoning makes sense so lets keep it for now, at least until we have more clarity about children values.